src/inference/graph_inference.cpp

```diff
@@ -341,6 +342,24 @@ class GraphNodeAssignment : public Nice2Assignment {
     return label_set_->GetLabelName(label_id);
   }
 
+  // Returns conditional probability of a label conditionated on its neighbouring labels
```

Comment:
I disagree a little bit on this, since if I include these two methods especially the one computing the normalisation factor then the PLLearn will become longer and I think less readable. In addition, I think the computation of the marginals and the estimation of the normalisation factor belongs more on a separated function than inside the learning procedure. However, if you think it would be more readable moving all the code into the PLLearn function I will do it, so let me know.